src/js/tech/middleware.js

```diff
@@ -15,8 +15,8 @@ export function getMiddleware(type) {
   return middlewares;
 }
 
-export function setSource(setTimeout, src, next) {
-  setTimeout(() => setSourceHelper(src, middlewares[src.type], next), 1);
+export function setSource(player, setTimeout, src, next) {
+  setTimeout(() => setSourceHelper(src, middlewares[src.type], next, player), 1);
```

Comment:
actually, I just realized I don't need to pass in `setTimeout` here. But it's the `setTimeout` from `player`, so, that if `player` is disposed`, so will this timeout.